diff -aurBb
--- S:/kabanov/mg1700/app/design/adminhtml/default/default/template/catalog/product/tab/inventory.phtml	Fri Apr 20 17:49:18 2012
+++ S:/kabanov/mg1700/app/design/adminhtml/default/default/template/catalog/product/tab/inventory-copy.phtml	Tue May 08 17:10:24 2012
@@ -37,6 +37,32 @@
 
         <legend><?php echo Mage::helper('catalog')->__('Inventory') ?></legend>
         <table cellspacing="0" class="form-list" id="table_cataloginventory">
+
+<?php /* START AITOC CODE */ ?>
+
+<?php if (!$this->IsNew() AND !$this->isDefaultWebsite()) { ?>
+
+        <tr>
+            <td class="label"><label for="inventory_use_default_website_stock"><?php echo Mage::helper('catalog')->__('Website Inventory') ?></label></td>
+            <td class="value">
+
+            <?php $_checked = ($this->getFieldValue('use_default_website_stock') || $this->IsNew()) ? 'checked="checked"' : '' ?>
+            <input type="hidden" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_default_website_stock]" value="0" />
+
+            <input type="checkbox" id="inventory_use_default_website_stock" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_default_website_stock]" value="1" <?php echo $_checked ?>  onclick="changeDefaultWebsiteStockOption();" class="checkbox" <?php echo $_readonly;?>/>
+            <label for="inventory_use_default_website_stock" class="normal"><?php echo Mage::helper('catalog')->__('Use Default Values') ?></label>
+
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
+        </tr>
+
+        <tr>
+            <td class="label" colspan="3"><br></td>
+        </tr>
+
+<?php } ?>
+
+<?php /* FINISH AITOC CODE */ ?>
+
         <tr>
             <td class="label"><label for="inventory_manage_stock"><?php echo Mage::helper('catalog')->__('Manage Stock') ?></label></td>
             <td class="value"><select id="inventory_manage_stock" name="<?php echo $this->getFieldSuffix() ?>[stock_data][manage_stock]" class="select" <?php echo $_readonly;?>>
@@ -49,7 +75,10 @@
             <input type="checkbox" id="inventory_use_config_manage_stock" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_config_manage_stock]" value="1" <?php echo $_checked ?> onclick="toggleValueElements(this, this.parentNode);" class="checkbox" <?php echo $_readonly;?>/>
             <label for="inventory_use_config_manage_stock" class="normal"><?php echo Mage::helper('catalog')->__('Use Config Settings') ?></label>
             <?php if (!$this->isReadonly()):?><script type="text/javascript">toggleValueElements($('inventory_use_config_manage_stock'), $('inventory_use_config_manage_stock').parentNode);</script><?php endif; ?></td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
 
 <?php if(!$this->getProduct()->isComposite()): ?>
@@ -61,7 +90,10 @@
                 <?php endif;?>
                 <input type="text" class="input-text required-entry validate-number" id="inventory_qty" name="<?php echo $this->getFieldSuffix() ?>[stock_data][qty]" value="<?php echo $this->getFieldValue('qty')*1 ?>" <?php echo $_readonly;?>/>
             </td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
 
         <tr>
@@ -72,7 +104,10 @@
             <input type="checkbox" id="inventory_use_config_min_qty" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_config_min_qty]" value="1" <?php echo $_checked ?> onclick="toggleValueElements(this, this.parentNode);" <?php echo $_readonly;?> />
             <label for="inventory_use_config_min_qty" class="normal"><?php echo Mage::helper('catalog')->__('Use Config Settings') ?></label>
             <?php if (!$this->isReadonly()):?><script type="text/javascript">toggleValueElements($('inventory_use_config_min_qty'), $('inventory_use_config_min_qty').parentNode);</script><?php endif; ?></td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
 
         <tr>
@@ -83,7 +118,10 @@
             <input type="checkbox" id="inventory_use_config_min_sale_qty" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_config_min_sale_qty]" value="1" <?php echo $_checked ?> onclick="toggleValueElements(this, this.parentNode);" class="checkbox" <?php echo $_readonly;?> />
             <label for="inventory_use_config_min_sale_qty" class="normal"><?php echo Mage::helper('catalog')->__('Use Config Settings') ?></label>
             <?php if (!$this->isReadonly()):?><script type="text/javascript">toggleValueElements($('inventory_use_config_min_sale_qty'), $('inventory_use_config_min_sale_qty').parentNode);</script><?php endif; ?></td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
 
         <tr>
@@ -94,7 +132,10 @@
             <input type="checkbox" id="inventory_use_config_max_sale_qty" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_config_max_sale_qty]" value="1" <?php echo $_checked ?> onclick="toggleValueElements(this, this.parentNode);" class="checkbox" <?php echo $_readonly;?> />
             <label for="inventory_use_config_max_sale_qty" class="normal"><?php echo Mage::helper('catalog')->__('Use Config Settings') ?></label>
             <?php if (!$this->isReadonly()):?><script type="text/javascript">toggleValueElements($('inventory_use_config_max_sale_qty'), $('inventory_use_config_max_sale_qty').parentNode);</script><?php endif; ?></td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
 
         <?php if($this->canUseQtyDecimals()): ?>
@@ -105,7 +146,10 @@
                 <option value="1"<?php if($this->getFieldValue('is_qty_decimal')==1): ?> selected="selected"<?php endif; ?>><?php echo Mage::helper('catalog')->__('Yes') ?></option>
             </select>
             </td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
 
         <?php if (!$this->isVirtual()) : ?>
@@ -116,7 +160,10 @@
                 <option value="1"<?php if($this->getFieldValue('is_decimal_divided') == 1): ?> selected="selected"<?php endif; ?>><?php echo Mage::helper('catalog')->__('Yes') ?></option>
             </select>
             </td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
         <?php endif; ?>
 
@@ -135,7 +182,10 @@
             <input type="checkbox" id="inventory_use_config_backorders" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_config_backorders]" value="1" <?php echo $_checked ?> onclick="toggleValueElements(this, this.parentNode);" class="checkbox" <?php echo $_readonly;?> />
             <label for="inventory_use_config_backorders" class="normal"><?php echo Mage::helper('catalog')->__('Use Config Settings') ?></label>
             <?php if (!$this->isReadonly()):?><script type="text/javascript">toggleValueElements($('inventory_use_config_backorders'), $('inventory_use_config_backorders').parentNode);</script><?php endif; ?></td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
         <tr>
             <td class="label"><label for="inventory_notify_stock_qty"><?php echo Mage::helper('catalog')->__('Notify for Quantity Below') ?></label></td>
@@ -145,7 +195,10 @@
             <input type="checkbox" id="inventory_use_config_notify_stock_qty" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_config_notify_stock_qty]" value="1" <?php echo $_checked ?> onclick="toggleValueElements(this, this.parentNode);" class="checkbox" <?php echo $_readonly;?>/>
             <label for="inventory_use_config_notify_stock_qty" class="normal"><?php echo Mage::helper('catalog')->__('Use Config Settings') ?></label>
             <?php if (!$this->isReadonly()):?><script type="text/javascript">toggleValueElements($('inventory_use_config_notify_stock_qty'), $('inventory_use_config_notify_stock_qty').parentNode);</script><?php endif; ?></td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
 <?php endif; ?>
         <tr>
@@ -160,7 +213,10 @@
             <input type="checkbox" id="inventory_use_config_enable_qty_increments" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_config_enable_qty_increments]" value="1" <?php echo $_checked ?> onclick="toggleValueElements(this, this.parentNode);" class="checkbox" <?php echo $_readonly;?>/>
             <label for="inventory_use_config_enable_qty_increments" class="normal"><?php echo Mage::helper('catalog')->__('Use Config Settings') ?></label>
             <?php if (!$this->isReadonly()):?><script type="text/javascript">toggleValueElements($('inventory_use_config_enable_qty_increments'), $('inventory_use_config_enable_qty_increments').parentNode);</script><?php endif; ?></td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
         <tr>
             <td class="label"><label for="inventory_qty_increments"><?php echo Mage::helper('catalog')->__('Qty Increments') ?></label></td>
@@ -170,7 +226,10 @@
                 <input type="checkbox" id="inventory_use_config_qty_increments" name="<?php echo $this->getFieldSuffix() ?>[stock_data][use_config_qty_increments]" value="1" <?php echo $_checked ?> onclick="toggleValueElements(this, this.parentNode);" class="checkbox" <?php echo $_readonly;?>/>
                 <label for="inventory_use_config_qty_increments" class="normal"><?php echo Mage::helper('catalog')->__('Use Config Settings') ?></label>
                 <?php if (!$this->isReadonly()):?><script type="text/javascript">toggleValueElements($('inventory_use_config_qty_increments'), $('inventory_use_config_qty_increments').parentNode);</script><?php endif; ?></td>
+            <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
         <tr>
             <td class="label"><label for="inventory_stock_availability"><?php echo Mage::helper('catalog')->__('Stock Availability') ?></label></td>
@@ -181,7 +240,10 @@
             <?php endforeach; ?>
             </select>
             </td>
+           <?php /* START AITOC COMMENTING
             <td class="value scope-label"><?php echo Mage::helper('adminhtml')->__('[GLOBAL]') ?></td>
+            FINISH AITOC COMMENTING */ ?>
+            <td class="value scope-label"><?php echo Mage::helper('catalog')->__('[WEBSITE]') ?></td>
         </tr>
         </table>
     </fieldset>
@@ -198,6 +260,7 @@
             inventory_max_sale_qty: true,
             inventory_enable_qty_increments : true,
             inventory_qty_increments: true
+            ,inventory_use_default_website_stock: true
         };
 
         $$('#table_cataloginventory tr').each(function(el) {
@@ -259,5 +322,70 @@
             applyEnableDecimalDivided();
         }
     });
+
+    <?php /* START AITOC CODE */ ?>
+
+    <?php if (!$this->IsNew() AND !$this->isDefaultWebsite()) { ?>
+
+    function changeDefaultWebsiteStockOption()
+    {
+        $$('#table_cataloginventory tr').each(manageAitocStockEach);
+
+        return true;
+    }
+
+    function manageAitocStockEach(el)
+    {
+        var catalogInventoryNotManagestockFields = {
+            inventory_min_sale_qty: true,
+            inventory_max_sale_qty: true,
+            inventory_enable_qty_increments : true,
+            inventory_qty_increments: true,
+            inventory_use_default_website_stock: true
+        };
+
+        if ($('inventory_use_default_website_stock').checked) {
+            var manageStock = 0;
+        }
+        else {
+            var manageStock = 1;
+
+            if ($('inventory_use_config_manage_stock').checked) {
+                manageStock = $('inventory_manage_stock_default').value;
+            }
+            else {
+                manageStock = $('inventory_manage_stock').value;
+            }
+
+            if (el == $('inventory_manage_stock').up(1)) {
+                manageStock = 1;
+            }
+
+            for(field in catalogInventoryNotManagestockFields) {
+                if ($(field) && ($(field).up(1) == el)) {
+                    manageStock = 1;
+                }
+            }
+
+        }
+        if ($("inventory_use_default_website_stock") && el == $('inventory_use_default_website_stock').up(1)) {
+            return;
+        }
+
+        if (manageStock == 1) {
+            el.show();
+        }
+        else {
+            el.hide();
+        }
+        return;
+    }
+
+    Event.observe(window, 'load', changeDefaultWebsiteStockOption);
+
+    <?php } ?>
+
+    <?php /* FINISH AITOC CODE */ ?>
+
 //]]>
 </script>
